### PR TITLE
Fix unsatisfiable constraints in MySite and Notification tabs

### PR DESCRIPTION
**Pull request by hassaanelgarem**
Part of #18098

## Description
This resolves a few unsatisfiable constraints errors related to the notifications and My Site. Errors fixed are:
- Issue in `NotificationsViewController`
    - This was triggered by a conflict of constraints while laying out the table's header view. 
    - Fixed by setting the filter bar height constraint priority to `999`
- Issue in `LatestPostSummaryCell`
    - This was triggered because the multiple stack views had leading and trailing constraints to superview. And this was conflicting with the width determined by its content.
    - Fixed by removing this constraint and instead, setting leading and trailing margins.
- Issue in `PostCardCell`
    - This was triggered by a conflict of constraints while laying out the cell. 
    - Fixed by setting the progress view height constraint priority to `999`
- Issue in `PageListTableViewCell`
    - This was triggered by a conflict of constraints while laying out the cell. The conflict came from auto resizing constraints on the content view
    - Fixed by setting `translatesAutoresizingMaskIntoConstraints` of the content view to false.
    
All fixes do not introduce any visible changes.

https://user-images.githubusercontent.com/25306722/157586738-065fdef7-6ec6-46ed-a750-24cdbb2df151.mov

## Testing Instructions
1. Open the notifications tab
2. ✅ Make sure that the header and filter bar is displayed correctly.
3. Open stats page and access the insights tab.
4. ✅ Make sure that the latest post summary is displayed correctly.
5. Open the Posts list screen
6. Make sure to enable the progress bar of any post, either normally or by overriding `PostCardStatusViewModel.shouldHideProgressView`
7. ✅ Make sure the cell is displayed correctly.
8. Open Pages list screen.
9. ✅ Make sure cells are displayed correctly.

## Regression Notes
1. Potential unintended areas of impact
UI of the components touched

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Testing manually all affected areas.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m [hunk 1]
```diff
@@ -33,7 +33,7 @@ @implementation PageListTableViewCell {
 - (void)awakeFromNib
 {
     [super awakeFromNib];
-
+    self.contentView.translatesAutoresizingMaskIntoConstraints = NO;
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m]: I'm getting these warnings:

```
2022-03-10 09:46:32.623389-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x14d0dba00; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12aa11160>>
2022-03-10 09:46:32.628937-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x12fbc9c00; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12b7a23b0>>
2022-03-10 09:46:32.633753-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x148135400; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x11ecb1fe0>>
2022-03-10 09:46:32.637881-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x1482cec00; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x11bfcbba0>>
2022-03-10 09:46:32.642078-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x13fb43400; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12dbda7e0>>
2022-03-10 09:46:32.646450-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x12fdbe200; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12b1905b0>>
2022-03-10 09:46:32.650387-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x12fe79400; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12aae61c0>>
2022-03-10 09:46:32.654155-0300 WordPress[9138:79856] [LayoutConstraints] Changing the translatesAutoresizingMaskIntoConstraints property of the contentView of a UITableViewCell is not supported and will result in undefined behavior, as this property is managed by the owning UITableViewCell. Cell: <PageListTableViewCell: 0x12f842200; baseClass = UITableViewCell; frame = (0 0; 320 60); autoresize = RM+BM; layer = <CALayer: 0x12b778cd0>>
```

**Review by leandroalonso — COMMENTED** [review]
There seems to be only an issue with the page cell. Other than that, all good visually!

[nit] In notifications when I change portrait/landscape there are a few constraint warnings being thrown.

File: WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m (hunk 1 above)
Comment (hassaanelgarem) [reply WordPress/Classes/ViewRelated/Pages/PageListTableViewCell.m]: Fixed! 93ab1d670394385fe158d4033fffcb3e9a1008a7

**Comment by hassaanelgarem** [comment]
> In notifications when I change portrait/landscape there are a few constraint warnings being thrown.

I tried to fix this but couldn't find a fast fix. I will try to fix this in another PR in the future.

**Review by leandroalonso — APPROVED** [review]
:shipit:

**Pull request merged**